SU2_CFD/include/output/COutput.hpp

```diff
@@ -0,0 +1,777 @@
+
+  unsigned short nDim;   /*!< \brief Physical Dimension */ 
+  
+  bool multiZone,       /*!< \brief Boolean to store whether we are running a multizone problem */ 
+  gridMovement,         /*!< \brief Boolean to store whether we have grid movement enabled */ 
+  femOutput;            /*!< \brief Boolean to store whether we should use the FEM routines */ 
+  
+  /*----------------------------- Screen and history output ----------------------------*/  
+  
+  string historySep;              /*!< \brief Character which separates values in the history file */  
+  unsigned short fieldWidth;      /*!< \brief Width of each column for the screen output (hardcoded for now) */
+  bool noWriting;                 /*!< \brief Boolean indicating whether a screen/history output should be written */
+  unsigned long curTimeIter,      /*!< \brief Current value of the time iteration index */
+  curAbsTimeIter,      /*!< \brief Current value of the time iteration index */  
+  curOuterIter,                   /*!< \brief Current value of the outer iteration index */
+  curInnerIter;                   /*!< \brief Current value of the inner iteration index */
+  
+  string historyFilename;       /*!< \brief The history filename*/
+  
+  /*! \brief Temporary variable to store the history filename */
+   char char_histfile[200];
+   /*! \brief Output file stream for the history */
+   ofstream histFile;
+  
+  /** \brief Enum to identify the screen output format. */
+  enum ScreenOutputFormat {           
+    FORMAT_INTEGER,         /*!< \brief Integer format. Example: 34 */
+    FORMAT_FIXED,           /*!< \brief Format with fixed precision for floating point values. Example: 344.54  */
+    FORMAT_SCIENTIFIC       /*!< \brief Scientific format for floating point values. Example: 3.4454E02 */  
+  };
+  
+  /** \brief Enum to identify the screen/history field type. */  
+  enum HistoryFieldType {           
+    TYPE_RESIDUAL,         /*!< \brief A user-defined residual field type*/
+    TYPE_AUTO_RESIDUAL,    /*!< \brief An automatically generated residual field type */
+    TYPE_COEFFICIENT,      /*!< \brief User defined coefficient field type  */
+    TYPE_AUTO_COEFFICIENT, /*!< \brief Automatically generated coefficient field type  */
+    TYPE_DEFAULT           /*!< \brief Default field type */  
+  };
+  
+  /** \brief Structure to store information for a history output field.
+   * 
+   *  The stored information is printed to the history file and to screen. 
+   * Each individual instance represents a single field (i.e. column) in the history file or on screen.   
+   */
+  struct HistoryOutputField {
+    /*! \brief The name of the field, i.e. the name that is printed in the screen or file header.*/
+    string              fieldName = ""; 
+    /*! \brief The value of the field. */
+    su2double           value = 0.0;
+    /*! \brief The format that is used to print this value to screen. */
+    unsigned short      screenFormat = FORMAT_FIXED; 
```

Comment:
if possible the enum type should be used here to.

c++ people also recommend using scoped enums (syntax `enum class`) as that guards against unintentional conversions from other types to enum, and makes intentional conversions clear (as an explicit cast is needed)

Reply:
Didn't know about the `enum class`. Seems like an appropriate way of doing it. Will be changed in the next commit. I am always learning new stuff! Nice